zio-http/src/main/scala/zhttp/http/URL.scala

```diff
@@ -3,20 +3,20 @@ package zhttp.http
 import java.net.{URI, URISyntaxException}
 import scala.util.Try
 
-case class URL(path: Path, kind: URL.Location = URL.Location.Relative) { self =>
+case class URL(path: Path, kind: URL.Location = URL.Location.Relative, query: String = "") { self =>
   val host: Option[String] = kind match {
-    case URL.Location.Relative             => None
-    case URL.Location.Absolute(_, host, _) => Option(host)
+    case URL.Location.Relative      => None
+    case abs: URL.Location.Absolute => Option(abs.host)
   }
 
   val port: Option[Int] = kind match {
-    case URL.Location.Relative             => None
-    case URL.Location.Absolute(_, _, port) => Option(port)
+    case URL.Location.Relative      => None
+    case abs: URL.Location.Absolute => Option(abs.port)
```

Comment:
Just out of curiosity is this needed?

Reply:
No, but it's more succinct when you have a growing number of fields.